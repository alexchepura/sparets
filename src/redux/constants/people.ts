export const REQUEST_PEOPLE = 'REQUEST_PEOPLE'
export const RECEIVE_PEOPLE = 'RECEIVE_PEOPLE'
