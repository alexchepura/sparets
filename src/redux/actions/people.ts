import { RECEIVE_PEOPLE, REQUEST_PEOPLE } from '../constants/people'

function requestPeople() {
  return {
    type: REQUEST_PEOPLE
  }
}

function receivePeople(json) {
  return {
    type: RECEIVE_PEOPLE,
    people: json.results
  }
}

export function fetchPeople() {
  return dispatch => {
    dispatch(requestPeople())
    return fetch(`https://randomuser.me/api/?results=3`)
      .then(response => response.json())
      .then(json => dispatch(receivePeople(json)))
  }
}
