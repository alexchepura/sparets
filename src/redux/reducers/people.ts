import { REQUEST_PEOPLE, RECEIVE_PEOPLE } from '../constants/people'

export function people(state: any = {
  items: [],
  isFetching: false
}, action) {
  switch (action.type) {
    case REQUEST_PEOPLE:
      return Object.assign({}, state, {
        isFetching: true
      })
    case RECEIVE_PEOPLE:
      return Object.assign({}, state, {
        isFetching: false,
        items: action.people
      })
    default:
      return state
  }
}
