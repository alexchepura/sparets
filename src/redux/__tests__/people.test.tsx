// import * as React from 'react'
// import * as renderer from 'react-test-renderer'
import React from 'react'
import renderer from 'react-test-renderer'

import { REQUEST_PEOPLE, RECEIVE_PEOPLE } from '../constants/people'
import { people } from '../reducers/people'
import PeopleComponent from '../components/People'

describe('people reducers', () => {
  it('changes isFetching to true', () => {
    const state = {
      isFetching: false
    }

    const result = people(state, {
      type: REQUEST_PEOPLE
    })

    expect(result.isFetching).toBeTruthy();
  })
})


describe('Snapshot People component', () => {
  it('renders simple list of people', () => {
    const peopleJSON = [{
      email: 'john@doe.com'
    },{
      email: 'jack@daniels.com'
    }]
    const component = renderer.create(
      <PeopleComponent
        people={peopleJSON}/>
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
