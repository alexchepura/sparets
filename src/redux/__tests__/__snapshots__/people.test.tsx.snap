exports[`Snapshot People component renders simple list of people 1`] = `
<ul>
  <li>
    john@doe.com
  </li>
  <li>
    jack@daniels.com
  </li>
</ul>
`;
